cupy/cuda/cutensor.pyx

```diff
@@ -171,39 +243,34 @@ cpdef inline check_status(int status):
 
 
 ###############################################################################
-# Handler creation/destruction
+# Handler initialization
 ###############################################################################
 
-cpdef size_t create() except? 0:
-    """Initializes the cuTensor library"""
-    cdef Handle handle
+cpdef size_t init() except? 0:
+    """Initializes the cuTENSOR library"""
+    cdef Handle *handle
+    _cutensor_alloc_handle(&handle)
```

Comment:
This memory is never freed.
How about managing the resource in RAII way?
I think you can allocate the memory as `std::shared_ptr<cutensorHandle_t>`, wrap it in some `cdef` class, and return it by value.

Reply:
I think what's done in `cupy/cuda/cupy_cutensor.cu` can totally be moved to `cupy/cuda/cutensor.pyx `. Note that in Cython you can do `from cpython.mem cimport PyMem_Malloc, PyMem_Free` for memory allocation: https://cython.readthedocs.io/en/latest/src/tutorial/memory_allocation.html

See also here: 
https://github.com/cupy/cupy/blob/5b06e239ec417a11bbc7743ad6af83088384df86/cupy/cuda/texture.pyx#L1-L2

Reply:
Thanks for your suggestion, @niboshi, @leofang.
I did not know how to allocate memory in Cython ^^;. It looks much better. I will fix the PR in that way.